runtime/protocol/http-client/common/src/aws/smithy/kotlin/runtime/http/interceptors/RequestCompressionInterceptor.kt

```diff
@@ -42,7 +44,9 @@ public class RequestCompressionInterceptor(
         }
 
         return if (algorithm != null && (context.protocolRequest.body.isStreaming || payloadSizeBytes?.let { it >= compressionThresholdBytes } == true)) {
-            algorithm.compressRequest(context.protocolRequest)
+            val request = algorithm.compressRequest(context.protocolRequest)
+            context.executionContext.emitBusinessMetric(BusinessMetrics.GZIP_REQUEST_COMPRESSION)
+            request
```

Comment:
**Style**: You can avoid this kind of temporary-storage-for-control-flow pattern by using [`also`](https://kotlinlang.org/docs/scope-functions.html#also):

```kotlin
algorithm.compressRequest(context.protocolRequest).also {
    context.executionContext.emitBusinessMetric(BusinessMetrics.GZIP_REQUEST_COMPRESSION)
}
```